### PR TITLE
Household create

### LINKED ISSUE #15: Welcom Page/Login
TBD on OAUTH

### PR DESCRIPTION
**Pull request by ztrokey**
# Description
Adds functionality to create households via API request.

# Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Closes #15 #12 

# How Has This Been Tested?
- [ ] This has not been tested
- [x] This has been tested

#Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes ***New spec is failing but should pass once user/dashboard is merged in.

### CODE REVIEW

File: app/poro/household.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Household
```
Comment (denverdevelopments) [inline app/poro/household.rb]: Good work.  I think the file naming convention is plural, app/poros/household.rb

File: app/views/households/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<h1>Household</h1>
+<br>
+<h4>Please create or join a household</h4>
```
Comment (denverdevelopments) [inline app/views/households/new.html.erb:3]: It seems like this is only the form to create new household, we may need a separate join.html.erb where they can join an existing household.

File: config/routes.rb [hunk 3]
```diff
@@ -3,5 +3,10 @@
   root 'welcome#index'
 
   get '/auth/:provider/callback', to: 'sessions#create'
+  namespace :user do
+    resources :dashboard, only: :show
+  end
   get '/dashboard', to: 'dashboard#show'
+
+  resources :households, only: %i[new create]
```
Comment (denverdevelopments) [inline config/routes.rb]: Will the household dashboard be the household show page?  Or is there where '/dashboard' redirects to, I forget?

**Review by denverdevelopments — APPROVED** [review]
Looks great Zach

File: app/controllers/households_controller.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        password: params[:password],
+        password_confirmation: params[:password_confirmation]
+      }
+    }
+
+    HouseholdsFacade.create_household(household_info)
+    flash[:success] = 'House Created!'
+    # redirect_to user_dashboard_path
+  end
+end
```
Comment (EmmyMorris) [inline app/controllers/households_controller.rb:20]: This looks amazing! Great Job with this create method!🎉

File: app/facades/households_facade.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class HouseholdsFacade
+  def self.create_household(household_info)
+    household_attributes = HouseholdService.create_or_find_household(household_info)
+    formatted = household_attributes[:data][:attributes]
+    Household.new(formatted)
+  end
+end
```
Comment (EmmyMorris) [inline app/facades/households_facade.rb:7]: Travis is giving us this error `uninitialized constant HouseholdsFacade::HouseholdService` on line 3 which is weird to me from looking this over.

File: app/services/household_service.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class HouseholdService
+  def self.connection
+    Faraday.new('https://choreganize-api.herokuapp.com/')
+  end
+
+  def self.create_or_find_household(household_info)
+    response = connection.post('/api/v1/household', household_info)
+    JSON.parse(response.body, symbolize_names: true)
+  end
+end
```
Comment (EmmyMorris) [inline app/services/household_service.rb:10]: This looks amazing!! Great job!

File: config/routes.rb (hunk 3 above)
Comment (EmmyMorris) [reply config/routes.rb]: Thats a good question! I know there is a household and a user page, I believe we can make them whatever seems to make most sense to us! Both of them as dashboards or both of them as shows?

File: spec/facades/households_facade_spec.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+      }
+    }
+    facade = HouseholdsFacade.create_household(household_info)
+
+    expect(facade).to be_an(Household)
+    expect(facade.address).to eq('123 Main Street')
+    expect(facade.city).to eq('Anytown')
+    expect(facade.state).to eq('CO')
+  end
+end
```
Comment (EmmyMorris) [inline spec/facades/households_facade_spec.rb:36]: Amazing job writing these test!! They look great!🎊

**Review by EmmyMorris — COMMENTED** [review]
This seems to have made travis quite mad! It seems like there are a few different errors that may need to be solved before travis will be happy. Other than travis, I think it looks really good! Also kevin's comment about a plural poros folder will definitely need to be made!

File: app/poro/household.rb (hunk 1 above)
Comment (suzkiee) [reply app/poro/household.rb]: oh yes, Kevin is right. It is plural!

File: config/routes.rb (hunk 3 above)
Comment (suzkiee) [reply config/routes.rb]: @denverdevelopments  yes, the household show page and the household dashboard page are the same! They should use the `/dashboard` route

**Review by EmmyMorris — APPROVED** [review]
Looks great!🤩

**Pull request merged**